test/lit/passes/inlining-eh-legacy.wast

```diff
@@ -109,36 +105,77 @@
 
   ;; CHECK:      (func $caller-with-pop (type $0)
   ;; CHECK-NEXT:  (local $0 i32)
+  ;; CHECK-NEXT:  (try
+  ;; CHECK-NEXT:   (do
+  ;; CHECK-NEXT:    (nop)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (catch $tag$0
+  ;; CHECK-NEXT:    (block $__inlined_func$callee-b$2
```

Comment:
How did this pass the validator? We haven't landed #6950...

Reply:
Yeah, this confused me too, I had to debug `eh-utils.cpp` for a bit...  the code there is careful, and if it sees a single block like this then it treats it as the block scope. So it is actually willing to allow a single block in such cases already, even without #6950, because in the binary format the body of a catch is a block scope anyhow that allows a list of instructions. And our binary writer will not emit an extra block there.

That actually means that #6950 is more of a gradual change actually... it just adds nameless blocks to the list of things we allow. Waybe we should reconsider landing it?

Reply:
Actually I realized that this happens to work out - it is also valid in the text format to not print such blocks. So we don't actually print them, and by skipping them, the wat text is valid.

That is both the binary and text format don't require an explicit block, and since they match, this is ok, and we never emit invalid wat.

Reply:
If we don't emit those nameless blocks in text, what was the problem that prevented #6950 from being landed? I might not have understood all discussions about the parser there...

Reply:
We don't emit a nameless block at a catch scope, but just one can be skipped:
```wat
(catch
  (block
    A
    B
  )
)
```
is the same as
```wat
(catch
  A
  B
)
```
which is valid wat. But
```wat
(catch
  (block
    (block
      A
      B
    )
    C
  )
)
```
would be
```wat
(catch
  (block
    A
    B
  )
  C
)
```
We can't avoid the inner block here in the text format, at least not in general (if it has a name and branches). Maybe we could extend the check to allow not just a single block but also nested blocks that can be merged, but that sounds less simple (and less worthwhile because this is legacy, as you said).